app/helpers/games_helper.rb

```diff
@@ -1,17 +1,38 @@
 module GamesHelper
-	def color_block(row, col)
-		if ((row % 2 == 0) && (col % 2 != 0)) || ((row % 2 != 0) && (col % 2 == 0))
-			"wht"
-		else
-			"blk"
-		end
-	end
+  def color_block(row, col)
+    if ((row % 2 == 0) && (col % 2 != 0)) || ((row % 2 != 0) && (col % 2 == 0))
+      "wht"
+    else
+      "blk"
+    end
+  end
 
-	def assign_piece(pieces, row, col)
-		pieces.each do |piece|
-			if (piece.row_position) == row && (piece.col_position == col)
-				"#{piece.type}"
-			end
-		end  
-	end 
-end
\ No newline at end of file
+  def assign_piece(type, color)
+    case
+    when type == 'Rook' && color == 'Black'
+      "black-rook"
+    when type == 'Knight' && color == 'Black'
+      "black-knight"
+    when type == 'Bishop' && color == 'Black'
+      "black-bishop"
+    when type == 'Queen' && color == 'Black'
+      "black-queen"
+    when type == 'King' && color == 'Black'
+      "black-king"
+    when type == 'Pawn' && color == 'Black'
+      "black-pawn"
+    when type == 'Rook' && color == 'White'
+      "white-rook"
+    when type == 'Knight' && color == 'White'
+      "white-knight"
+    when type == 'Bishop' && color == 'White'
+      "white-bishop"
+    when type == 'Queen' && color == 'White'
+      "white-queen"
+    when type == 'King' && color == 'White'
+      "white-king"
+    when type == 'Pawn' && color == 'White'
+      "white-pawn"
+    end
+  end
```

Comment:
is there a way you could make this more dynamic?  Think about string interpolation e.g. "#{my_color_var}-#{my_piece_type_var}" - it moves this to 1 line of code that isn't prone to typo errors.  Just make sure to .downcase your variables.  Also, this is a great bit of code to test.